public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -507,13 +506,10 @@ export class DashboardModel implements TimeModel {
       return;
     }
 
-    this.iteration = (this.iteration || new Date().getTime()) + 1;
     // cleanup scopedVars
     deleteScopeVars(this.panels);
 
-    const panelsToRemove = this.panels.filter(
-      (p) => (!p.repeat || p.repeatedByRow) && p.repeatPanelId && p.repeatIteration !== this.iteration
```

Comment:
I don't see any scenario where `p.repeatIteration !== this.iteration` would ever be false, all my testing it's always true